File: fern/snippets/tool-router/python/openai-agents.py
from composio import Composio
from composio_openai_agents import OpenAIAgentsProvider
from agents import Agent, Runner

composio = Composio(
    api_key="your-api-key",
    provider=OpenAIAgentsProvider()
)

session = composio.create(user_id="pg-user-550e8400-e29b-41d4")

agent = Agent(
    name="Personal Assistant",
    instructions="You are a helpful personal assistant.",
    tools=session.tools()
)

result = Runner.run_sync(
    agent,
    "Summarize all the emails in my Gmail inbox today"
)
print(result.final_output)

File: fern/snippets/tool-router/typescript/openai-agents.ts
import { Composio } from "@composio/core";
import { OpenAIAgentsProvider } from "@composio/openai-agents";
import { Agent, run } from "@openai/agents";

const composio = new Composio({
  apiKey: "your-api-key",
  provider: new OpenAIAgentsProvider(),
});

const { tools } = await composio.create("pg-user-550e8400-e29b-41d4");

const agent = new Agent({
  name: "Personal Assistant",
  instructions: "You are a helpful personal assistant.",
  tools: await tools(),
});

const result = await run(
  agent,
  "Summarize all the emails in my Gmail inbox today"
);
console.log(result.finalOutput);
